boards.txt

```diff
@@ -2014,88 +2012,88 @@ LoRa.menu.upload_method.dfuMethod.upload.tool=stm32CubeProg
 
 
 ###############################
-# Genericflight Flight Controllers
+# Generic Flight Controllers
 
-Genericflight.name=Generic flight controllers
+GenFlight.name=Generic Flight Controllers
```

Comment:
FYI, this board ID change is breaking because it changes the FQBN used to identify the board via the command line (both for Arduino IDE and Arduino CLI).

Before: `STM32:stm32.Genericflight`
After: ``STM32:stm32.GenFlight`

Reply:
Yes but I see no issue. It changes that's all or maybe I missed something ?

Reply:
Well, I consider the FQBN part of the API of a boards platform, and thus cause for a major version bump.

Anyone who tries to use the old FQBN in a command will find it's now broken:

```
$ arduino-cli compile --fqbn STM32:stm32.Genericflight
Error during build: Error resolving FQBN: board STM32:stm32.Genericflight@1.9.1:Genericflight not found
```

This is especially problematic for automated systems like CI.

If you think it should be changed, fine, but I just want you to be aware that this isn't just cosmetic. It should certainly be prominently documented in the changelog.

Reply:
OK. Yes there will be a major version bum (2.0.0). I will also change the maintainer name as STM32 is not a vendor nor a maintainer and of course all will be documented in the changelog.

FQBN will looks like: `STMicroelectronics:stm32`

For our CI the change will not change anything as the script get the fqbn dynamically.

Thanks for your feedback.

Reply:
A non-breaking option would be to make a copy of the board definition, so you have one definition with the deprecated `Genericflight` board ID and another with the new `GenFlight` ID, then add a `hide` property to the deprecated board definition:
https://arduino.github.io/arduino-cli/latest/platform-specification/#hiding-boards
```
Genericflight.hide=
```
This will cause the Arduino IDE to not display the deprecated board in its **Tools > Board** menu. The `hide` property is not yet supported by Arduino CLI or Arduino Pro IDE, so they will still show the deprecated board option.

Reply:
Thanks for pointing this, anyway I don't think the `hide` option is required in this case.